src/idl_parser.cpp

```diff
@@ -633,7 +633,7 @@ CheckedError Parser::ParseField(StructDef &struct_def) {
   field->deprecated = field->attributes.Lookup("deprecated") != nullptr;
   auto hash_name = field->attributes.Lookup("hash");
   if (hash_name) {
-    switch (type.base_type) {
+    switch (IsScalar(type.base_type) ? type.base_type : type.element) {
```

Comment:
should we check here that if it is not scalar, it is actually a vector?

Reply:
fair point.
I reversed the logic to check for BASE_TYPE_VECTOR instead of IsScalar.